src/control/tools/StrokeHandler.cpp

```diff
@@ -58,24 +62,72 @@ auto StrokeHandler::onMotionNotifyEvent(const PositionInputData& pos) -> bool {
         return false;
     }
 
-    double zoom = xournal->getZoom();
-    double x = pos.x / zoom;
-    double y = pos.y / zoom;
-    int pointCount = stroke->getPointCount();
+    if (pos.pressure == 0) {
+        /**
+         * Some devices emit a move event with pressure 0 when lifting the stylus tip
+         * Ignore those events
+         */
+        return true;
+    }
 
-    Point currentPoint(x, y);
+    stabilizer->processEvent(pos);
+    return true;
+}
+
+void StrokeHandler::paintTo(const Point& point) {
+
+    int pointCount = stroke->getPointCount();
 
     if (pointCount > 0) {
-        if (!validMotion(currentPoint, stroke->getPoint(pointCount - 1))) {
-            return true;
+        Point endPoint = stroke->getPoint(pointCount - 1);
+        double distance = point.lineLengthTo(endPoint);
+        if (distance < PIXEL_MOTION_THRESHOLD) {  //(!validMotion(point, endPoint)) {
```

Comment:
Is `distance` used after the if? If not, include it into the ifStmt.

```suggestion
        if (double distance = point.lineLengthTo(endPoint); distance < PIXEL_MOTION_THRESHOLD) {  //(!validMotion(point, endPoint)) {
```

Reply:
Yes, it is used below (line 96)